feat(external docs): Helm chart docs updates - reorging and adding Aggregator docs (#8824)
timberio/vector: website/content/en/docs/setup/installation/package-managers/helm.md

weight: 3
---

[Helm] is a package manager for Kubernetes that facilitates the deployment and management of applications and services on Kubernetes clusters. This page covers installing and managing the Vector Agent and the Vector Aggregator through the Helm package repository. The Agent and the Aggregator are added by adding the Vector Helm chart but are installed onto each node and configured separately.

> {{< warning title="Aggregator role in private beta" >}}

Author: ```suggestion
{{< warning title="Aggregator role in public beta" >}}
```

> Helm support for the [aggregator] role is currently in private beta. We're currently seeking beta testers. If interested, please [join our chat][chat] and let us know.

Author: ```suggestion
Helm support for the [aggregator] role is currently in public beta. We're currently seeking beta testers. If interested, please [join our chat][chat] and let us know.
```

[agent]: /docs/setup/deployment/roles/#agent
[aggregator]: /docs/setup/deployment/roles/#aggregator
[chat]: https://chat.vector.dev
{{< /warning >}}

## Adding the Helm repo

If you haven't already, start by adding the Vector repo:

```shell
helm repo add timberio https://packages.timber.io/helm/latest
helm repo update
```

## Agent

The Vector [Agent] lets you collect data from your [sources] and then deliver it to a variety of destinations with [sinks].

### Installing

Once you add the Vector Helm repo, install the Vector Agent to each node:

```shell
helm install vector timberio/vector-agent \
  --namespace vector \
  --create-namespace \

>   --values values.yaml

Reviewer: One note is the user may not have a local `values.yaml` until we have them generate the example on ln 63.

`helm install` without the `-f`/`--values` option would use our packaged configuration which is valid, but doesn't do much today.

Author: okay, that makes a lot more sense. I will swap them in both!

```

### Updating

Or to update the Vector Agent:

```shell
helm repo update && \
helm upgrade vector timberio/vector-agent \
  --namespace vector \
  --reuse-values
```

### Configuring

To check available Helm chart configuration options:

```shell
helm show values timberio/vector-agent
```

This example configuration file lets you use Vector as an Agent to send logs to standard output. For more information about configuration options, see the [configuration] docs page.

> ```toml

Reviewer: ```suggestion
```yaml
```
🤔 I suspect we can change this since the snippet is (a shell command) essentially a YAML snippet.

weight: 3
---

> [Helm] is a package manager for Kubernetes that facilitates the deployment and management of applications and services on Kubernetes clusters. This page covers installing and managing the Vector Agent and the Vector Aggregator from the Helm package repository. The Agent and the Aggregator are added by adding the Vector Helm chart but are installed and configured separately.

Author: yeah, I see the distinction you are making and I think that makes it much more clear

{{< warning title="Aggregator role in public beta" >}}

> Helm support for the [aggregator] role is currently in public beta. We're currently seeking beta testers. If interested, please [join our chat][chat] and let us know.

Reviewer 2: ```suggestion
Helm support for the [aggregator] role is currently in public beta. We're seeking beta testers! If deploying the aggregator chart, please [join our chat][chat] and let us know how it went.
```

[aggregator]: /docs/setup/deployment/roles/#aggregator
[chat]: https://chat.vector.dev
{{< /warning >}}

## Adding the Helm repo

If you haven't already, start by adding the Vector repo:

```shell

> helm repo add timberio https://packages.timber.io/helm/latest

Reviewer 2: Is this right, shouldn't it be `helm repo add vector ...` cc/ @spencergilbert

Reviewer: Both are supported for this release - we should probably make sure it's consistent between this page and the highlight

Reviewer 2: 👍 do you have a preference? I'm inclined to use the new repo since these are new docs and they assume you are starting from scratch.

Reviewer: Sure - we can keep the `helm repo add vector https://helm.vector.dev` and sed timberio/vector

VALUES
```

### Installing

Once you add the Vector Helm repo, and added a Vector configuration file, install the Vector Agent:

```shell
helm install vector timberio/vector-agent \
  --namespace vector \
  --create-namespace \
  --values values.yaml
```

### Updating

Or to update the Vector Agent:

```shell
helm repo update && \
helm upgrade vector timberio/vector-agent \
  --namespace vector \
  --reuse-values
```

## Aggregator

> The Vector [Aggregator] lets you [transform] your data. For example, dedupe, aggregate, or redact the data before sending it to its final destination.

Reviewer 2: I might copy the aggregator description @spencergilbert has over here: https://github.com/timberio/vector/pull/8800/files#diff-8a968d252761d4b71779d1117a16b224977b3bad224511172b871192f86617adR26-R33 . I think that one has a bit more color.

Author: I was trying to keep it more minimal than the big description (though I do agree that one is good). Would this be a good middle ground:

```
The Vector [Aggregator] lets you [transform] and ship data collected by other agents. For example, it can insure that the data you are collecting is scrubbed of sensitive information, properly formatted for downstream consumers, sampled to reduce volume, and more.
```

Reviewer 2: 👍 good point, it does sense to keep it more concise on this page. I like your suggestion.

### Configuring

To check available Helm chart configuration options:

```shell
helm show values timberio/vector-aggregator
```

This example configuration file lets you use Vector as an Aggregator to parse events to make them human-readable. For more information about configuration options, see the [Configuration] docs page.

```yaml
cat <<-'VALUES' > values.yaml
# The Vector Aggregator chart defines a
# vector source that is made available to you.
# You do not need to define a log source.
transforms:
  # Adjust as necessary. This remap transform parses a JSON
  # formatted log message, emitting a log if the contents are
  # not valid JSON
  # /docs/reference/transforms/
  remap:
    type: remap
    inputs: ["vector"]
    source: |
      structured, err = parse_json(.message)
      if err != null {
        log("Unable to parse JSON: " + err, level: "error")
      } else {
        . = merge(., object!(structured))
      }
sinks:
  # Adjust as necessary. By default we use the console sink
  # to print all data. This allows you to see Vector working.
  # /docs/reference/sinks/
  stdout:
    type: console
    inputs: ["remap"]
    target: "stdout"
    encoding: "json"

> VALUES

Reviewer: Missing the ``` here!

> VALUES

Author: ```suggestion
VALUES
```
```

### Installing

Once you add the Vector Helm repo, and add a Vector configuration file, install the Vector Aggregator:

```shell
helm install vector timberio/vector-aggregator \
  --namespace vector \
  --create-namespace \
  --values values.yaml
```

### Updating

Or to update the Vector Aggregator:

```shell
helm repo update && \
helm upgrade vector timberio/vector-aggregator \
  --namespace vector \
  --reuse-values
```

## Uninstalling Vector

To uninstall the Vector helm chart:

```shell
helm uninstall vector --namespace vector 
```

## Management

{{< jump "/docs/administration/management" "helm" >}}

[helm]: https://helm.sh
[Configuration]: /docs/reference/configuration/
[Agent]: /docs/setup/deployment/roles/#agent
[sources]: /docs/reference/configuration/sources/
[sinks]: /docs/reference/configuration/sinks/

> [Aggregator]: /docs/setup/deployment/roles/#aggregator

Author: ```suggestion
[Aggregator]: /docs/setup/deployment/roles/#aggregator
[transform]: /docs/reference/configuration/transforms/
```

VALUES
```

### Installing

Once you add the Vector Helm repo, and added a Vector configuration file, install the Vector Agent:

```shell
helm install vector timberio/vector-agent \
  --namespace vector \
  --create-namespace \
  --values values.yaml
```

### Updating

Or to update the Vector Agent:

```shell
helm repo update && \
helm upgrade vector timberio/vector-agent \
  --namespace vector \
  --reuse-values
```

## Aggregator

The Vector [Aggregator] lets you [transform] your data. For example, dedupe, aggregate, or redact the data before sending it to its final destination.

### Configuring

To check available Helm chart configuration options:

```shell
helm show values timberio/vector-aggregator
```

This example configuration file lets you use Vector as an Aggregator to parse events to make them human-readable. For more information about configuration options, see the [Configuration] docs page.

> ```toml
> # Add example
> ```

Reviewer: ```suggestion
```yaml
cat <<-'VALUES' > values.yaml
# The Vector Aggregator chart defines a vector
# kubernetes_logs source that is made available to you.
# You do not need to define a log source.
transforms:
  # Adjust as necessary. This remap transform parses a JSON
  # formatted log message, emitting a log if the contents are
  # not valid JSON
  # /docs/reference/transforms/
  remap:
    type: remap
    inputs: ["vector"]
    source: |
      structured, err = parse_json(.message)
      if err != null {
        log("Unable to parse JSON: " + err, level: "error")
      } else {
        . = merge(., structured)
      }
sinks:
  # Adjust as necessary. By default we use the console sink
  # to print all data. This allows you to see Vector working.
  # /docs/reference/sinks/
  stdout:
    type: console
    inputs: ["remap"]
    target: "stdout"
    encoding: "json"
VALUES
```
```

@jszwedko @binarylogic does this work as the quickstart config, or do we want to coalesce a number of parse functions?

Reviewer 2: Should ` . = merge(., structured)` be ` . = merge(., object!(structured))`? I see:

```
  │
5 │                             . = merge(., structured)
  │                             --- ^^^^^^^^^^^^^^^^^^^^
  │                             │   │
  │                             │   this expression is fallible
  │                             │   update the expression to be infallible
  │                             or change this to an infallible assignment:
  │                             ., err = merge(., structured)
```

running this config. Alternatively, we could handle that error separately.

I think this config as an example makes sense though.

Reviewer: Doh

Reviewer 2: Looks good to me 👍
